Source/WebCore/accessibility/AXObjectCache.cpp

```diff
@@ -4658,44 +4659,9 @@ void AXObjectCache::deferTextReplacementNotificationForTextControl(HTMLTextFormC
     m_deferredTextFormControlValue.add(formControlElement, previousValue);
 }
 
-bool isNodeAriaVisible(Node* node)
-{
-    return node ? isNodeAriaVisible(*node) : false;
-}
-
-bool isNodeAriaVisible(Node& node)
+bool isNodeFocused(Node& node)
 {
-    // If an element is focused, it should not be hidden.
-    if (RefPtr element = dynamicDowncast<Element>(node); element && element->focused())
-        return true;
-
-    // ARIA Node visibility is controlled by aria-hidden
-    //  1) if aria-hidden=true, the whole subtree is hidden
-    //  2) if aria-hidden=false, and the object is rendered, there's no effect
-    //  3) if aria-hidden=false, and the object is NOT rendered, then it must have
-    //     aria-hidden=false on each parent until it gets to a rendered object
-    //  3b) a text node inherits a parents aria-hidden value
-    bool requiresAriaHiddenFalse = !node.renderer();
-    bool ariaHiddenFalsePresent = false;
-    for (auto* testNode = &node; testNode; testNode = testNode->parentNode()) {
-        if (RefPtr element = dynamicDowncast<Element>(*testNode)) {
-            const auto& ariaHiddenValue = element->attributeWithoutSynchronization(aria_hiddenAttr);
-            if (equalLettersIgnoringASCIICase(ariaHiddenValue, "true"_s))
-                return false;
-
-            // We should break early when it gets to the body.
-            if (testNode->hasTagName(bodyTag))
-                break;
-
-            bool ariaHiddenFalse = equalLettersIgnoringASCIICase(ariaHiddenValue, "false"_s);
-            if (!testNode->renderer() && !ariaHiddenFalse)
-                return false;
-            if (!ariaHiddenFalsePresent && ariaHiddenFalse)
-                ariaHiddenFalsePresent = true;
-        }
-    }
-    
-    return !requiresAriaHiddenFalse || ariaHiddenFalsePresent;
+    return is<Element>(node) && uncheckedDowncast<Element>(node).focused();
```

Comment:
Can we remove the aria-hidden=true check in the walk up?

Reply:
I think I'll need to add that back in (probably in `isARIAHidden()`). Doesn't look like we have a test covering that, so I'll add that in to my new test as well.

Reply:
I addressed this in the latest revision by using `isAXHidden()` (with the appropriate inverse logic) in places where we used to do `isNodeARIAVisible()`. `isAXHidden()` takes care of the ancestor walk-up to check for ARIA hidden.

When computing `defaultObjectInclusion()`, however, we already perform an ancestor walk to check for `aria-hidden=true`, so that case is covered already:
```    
    if (isARIAHidden())
        return AccessibilityObjectInclusion::IgnoreObject;

    bool ignoreARIAHidden = isFocused();
    if (Accessibility::findAncestor<AccessibilityObject>(*this, false, [&] (const auto& object) {
        return (!ignoreARIAHidden && object.isARIAHidden()) || object.ariaRoleHasPresentationalChildren() || !object.canHaveChildren();
    }))
        return AccessibilityObjectInclusion::IgnoreObject;
 ```